lab_1/main.py

```diff
@@ -86,10 +160,37 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
+
+    common = []
+    if not (isinstance(unknown_profile, dict) and isinstance(profile_to_compare, dict)
+            and isinstance(top_n, int)):
+        return None
+    tokens = list(profile_to_compare.get('freq').keys())
+    max_length_word = max(tokens, key=len)
```

Comment:
explain

Reply:
it searches for the longest word in the list of tokens by using the max function with key=len so it would use word length as a measure